File: BackEnd/Domain/Aggregates/Project.cs
using SharedKernel.Aggregate;
using SharedKernel.Constants;

namespace Domain.Aggregates;

public class Project : AggregateRoot<Guid>
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public ProjectStatus Status { get; set; } = ProjectStatus.Active;
    public DateTime? StartDate { get; set; } = DateTime.Now;
    public DateTime? EndDate { get; set; } = DateTime.Now.AddYears(1);
    
    public ICollection<User> Users { get; set; } = new List<User>();
    public ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
    public ICollection<TicketStatus> TicketStatuses { get; set; } = new List<TicketStatus>();
    
    public static Project Create(string name, string description, string code, DateTime? startDate = null, DateTime? endDate = null)
    {
        return new Project
        {
            Id = Guid.CreateVersion7(),
            Name = name,
            Description = description,
            Code = code,
            Status = ProjectStatus.Active,
            StartDate = startDate,
            EndDate = endDate
        };
    }

    public void SetUsers(List<User> users)
    {
        Users = users;
    }
    
    public void Update(string name, string description, string code, DateTime? startDate = null, DateTime? endDate = null)
    {
        Name = name;
        Description = description;
        Code = code;
        StartDate = startDate;
        EndDate = endDate;
        
    }
}

File: flutter/lib/domain/aggregates/project/project.dart
import 'package:dart_mappable/dart_mappable.dart';
import 'package:enterprise_management/domain/aggregates/ticket_status/ticket_status.dart';
import 'package:enterprise_management/domain/aggregates/user/user.dart';
import 'package:enterprise_management/shared_kernel/aggregate/aggregate_root.dart';

part 'project.mapper.dart';

@MappableClass()
class Project extends AggregateRoot<String> with ProjectMappable {
  Project({
    required super.id,
    required this.name,
    required this.description,
    this.isActive = false,
    required this.code,
    required this.status,
    required this.startDate,
    required this.endDate,
    required this.users,
    required this.ticketStatuses,
  });

  final String name;
  final String description;
  final String code;
  final int status;
  final DateTime? startDate;
  final DateTime? endDate;
  final List<User> users;
  final List<TicketStatus> ticketStatuses;
  final bool isActive;
}
